runtime/src/main/java/org/corfudb/protocols/logprotocol/SMREntry.java

```diff
@@ -117,6 +120,30 @@ void deserializeBuffer(ByteBuf b, CorfuRuntime rt) {
         SMRArguments = arguments;
     }
 
+    /**
+     * Given a buffer with the reader index pointing to a serialized SMREntry, this method will
+     * seek the buffer's reader index to the end of the entry.
+     */
+    public static void seekToEnd(ByteBuf b) {
+        // Magic
+        byte magicByte = b.readByte();
+        checkState(magicByte == CorfuSerializer.corfuPayloadMagic, "Not a ICorfuSerializable object");
```

Comment:
This doesn't look like a clean implementation under the current abstraction. I think we should do this inside CorfuSerializer (add a method in CorfuSerializer), and not exposing the handling of the magic.

Reply:
Good point. I'll just add a seek method to ISerializer and change this to private. We shouldn't expose this.

Reply:
> This seems like a short-term solution right? Because all these seeks might incur performance regression.

Unless we change the format (i.e. prefix the whole container's length with a length field), we can't just skip bytes, and the point of this patch is to not change the format. Let me know if you have a more optimal way.

Reply:
@WenbinZhu @xnull By hiding the magic byte, I need to make a couple of components "seek aware", given that this is only a short term solution I think this patch is a reasonable one. 

I tried to the other implementation here:
https://github.com/CorfuDB/CorfuDB/compare/multiObjectSMRFormat2

Also, once we change the format, this won;t be needed, so I'm kind of hesitant to make the uber change.